### PR TITLE
fix(anta.tests): use json outformat in VerifyInterfaceUtilization

### PR DESCRIPTION
**Pull request by mtache**
- fix(anta.tests): use json outformat in VerifyInterfaceUtilization
- test(anta.tests): update VerifyInterfaceUtilization unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/interfaces.py [hunk 1]
```diff
@@ -26,35 +26,62 @@
...
     name = "VerifyInterfaceUtilization"
-    description = "Verifies that all interfaces have a usage below 75%."
+    description = "Verifies interfaces utilization is below a threshold."
     categories = ["interfaces"]
-    # TODO - move from text to json if possible
-    commands = [AntaCommand(command="show interfaces counters rates", ofmt="text")]
+    commands = [AntaCommand(command="show interfaces counters rates"), AntaCommand(command="show interfaces")]
+
+    class Input(AntaTest.Input):
+        """Input for the VerifyInterfaceUtilization test."""
```
Comment (gmuloc) [inline anta/tests/interfaces.py:43]: please add input example here :P jk

File: anta/tests/routing/bgp.py [hunk 2]
```diff
@@ -11,7 +11,8 @@
 from ipaddress import IPv4Address, IPv4Network, IPv6Address
 from typing import Any, List, Optional, Union, cast
 
-from pydantic import BaseModel, Field, PositiveInt, model_validator, utils
+from pydantic import BaseModel, Field, PositiveInt, model_validator
+from pydantic.v1.utils import deep_update
```
Comment (gmuloc) [inline anta/tests/routing/bgp.py:15]: can we avoid this?

Comment (mtache) [reply anta/tests/routing/bgp.py:15]: Pydantic sends a warning asking to change the import: https://docs.pydantic.dev/latest/migration/

File: anta/tests/interfaces.py (hunk 1 above)
Comment (mtache) [reply anta/tests/interfaces.py:43]: later :P

File: anta/tests/interfaces.py [hunk 3]
```diff
@@ -26,35 +26,62 @@
...
+
+        def _test_intf(intf: str) -> None:
+            bandwidth = interfaces["interfaces"][intf]["bandwidth"]
+
+            def _test_rate(rate: str) -> None:
+                usage = rates[rate] / bandwidth * 100
+                if usage > self.inputs.threshold:
+                    if intf not in failed_interfaces:
+                        failed_interfaces[intf] = {}
+                    failed_interfaces[intf][rate] = usage
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:63]:
```suggestion
                    failed_interfaces.setdefault(intf, {})[rate] = usage
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM if CI passes!

File: anta/tests/routing/bgp.py (hunk 2 above)
Comment (gmuloc) [reply anta/tests/routing/bgp.py:15]: yeah we may need to keep an eye on this one..

**Pull request merged**